contracts/traderjoe/FarmLensV2.sol

```diff
@@ -0,0 +1,364 @@
+
+    function joePerSec() external view returns (uint256);
+}
+
+contract FarmLens {
+    using SafeMath for uint256;
+
+    struct FarmPair {
+        uint256 id;
+        uint256 allocPoint;
+        address lpAddress;
+        address token0Address;
+        address token1Address;
+        string token0Symbol;
+        string token1Symbol;
+        uint256 reserveUsd;
+        uint256 totalSupplyScaled;
+        address chefAddress;
+        uint256 chefBalanceScaled;
+        uint256 chefTotalAlloc;
+        uint256 chefJoePerSec;
+    }
+
+    struct BMCJInfo {
+        uint256 baseJoePerYear;
+        uint256 boostedJoePerYear;
+        uint256 boostFactor;
+    }
+
+    struct AllFarmData {
+        uint256 avaxPriceUsd;
+        uint256 joePriceUsd;
+        uint256 totalAllocChefV2;
+        uint256 totalAllocChefV3;
+        uint256 totalAllocBMCJ;
+        uint256 joePerSecChefV2;
+        uint256 joePerSecChefV3;
+        uint256 joePerSecBMCJ;
+        FarmPair[] farmPairsV2;
+        FarmPair[] farmPairsV3;
+        FarmPair[] farmPairsBMCJ;
+    }
+
+    address public immutable joe; // 0x6e84a6216eA6dACC71eE8E6b0a5B7322EEbC0fDd;
+    address public immutable wavax; // 0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7;
+    address public immutable wavaxUsdte; // 0xeD8CBD9F0cE3C6986b22002F03c6475CEb7a6256
+    address public immutable wavaxUsdce; // 0xA389f9430876455C36478DeEa9769B7Ca4E3DDB1
+    address public immutable wavaxUsdc; // 0xf4003f4efbe8691b60249e6afbd307abe7758adb
+    IJoeFactory public immutable joeFactory; // 0x9Ad6C38BE94206cA50bb0d90783181662f0Cfa10
+    IMasterChef public immutable chefv2; // 0xd6a4F121CA35509aF06A0Be99093d08462f53052
+    IMasterChef public immutable chefv3; // 0x188bED1968b795d5c9022F6a0bb5931Ac4c18F00
+    IBoostedMasterchef public immutable bmcj; // Not Deplyed Yet
+
+    constructor(
+        address _joe,
+        address _wavax,
+        address _wavaxUsdte,
+        address _wavaxUsdce,
+        address _wavaxUsdc,
+        IJoeFactory _joeFactory,
+        IMasterChef _chefv2,
+        IMasterChef _chefv3,
+        IBoostedMasterchef _bmcj
+    ) public {
+        joe = _joe;
+        wavax = _wavax;
+        wavaxUsdte = _wavaxUsdte;
+        wavaxUsdce = _wavaxUsdce;
+        wavaxUsdc = _wavaxUsdc;
+        joeFactory = _joeFactory;
+        chefv2 = _chefv2;
+        chefv3 = _chefv3;
+        bmcj = _bmcj;
+    }
+
+    /// @notice Returns price of avax in usd.
+    function getAvaxPrice() public view returns (uint256) {
+        uint256 priceFromWavaxUsdte = _getAvaxPrice(IJoePair(wavaxUsdte)); // 18
+        uint256 priceFromWavaxUsdce = _getAvaxPrice(IJoePair(wavaxUsdce)); // 18
+        uint256 priceFromWavaxUsdc = _getAvaxPrice(IJoePair(wavaxUsdc)); // 18
+
+        uint256 sumPrice = priceFromWavaxUsdte.add(priceFromWavaxUsdce).add(
+            priceFromWavaxUsdc
+        ); // 18
+        uint256 avaxPrice = sumPrice / 3; // 18
```

Comment:
Safemath for div is really useless tbh, it can't overflow nor underflow.
The only use is that if `a/b` b could be 0. (in some old solidity versions, 1/0 would return max(uint256), but it seems like in solidity 0.6 it reverts)